src/Aspire.Hosting/Dashboard/DashboardServiceHost.cs

```diff
@@ -68,36 +75,64 @@ internal sealed class DashboardServiceHost : IHostedService
 
         return;
 
-        static void ConfigureKestrel(KestrelServerOptions kestrelOptions)
+        void ConfigureKestrel(KestrelServerOptions kestrelOptions)
         {
-            var uris = EnvironmentUtil.GetAddressUris(DashboardServiceUrlVariableName, DashboardServiceUrlDefaultValue);
 
-            foreach (var uri in uris)
+            if (DashboardServiceUri.IsLoopback)
             {
-                if (uri.IsLoopback)
-                {
-                    kestrelOptions.ListenLocalhost(uri.Port, ConfigureListen);
-                }
-                else
-                {
-                    kestrelOptions.Listen(IPAddress.Parse(uri.Host), uri.Port, ConfigureListen);
-                }
+                kestrelOptions.ListenLocalhost(dashboardServiceUri.Port, ConfigureListen);
+            }
+            else
+            {
+                kestrelOptions.Listen(IPAddress.Parse(DashboardServiceUri.Host), DashboardServiceUri.Port, ConfigureListen);
+            }
 
-                void ConfigureListen(ListenOptions options)
+            void ConfigureListen(ListenOptions options)
+            {
+                // Force HTTP/2 for gRPC, so that it works over non-TLS connections
+                // which cannot negotiate between HTTP/1.1 and HTTP/2.
+                options.Protocols = HttpProtocols.Http2;
+
+                if (string.Equals(DashboardServiceUri.Scheme, "https", StringComparison.Ordinal))
                 {
-                    // Force HTTP/2 for gRPC, so that it works over non-TLS connections
-                    // which cannot negotiate between HTTP/1.1 and HTTP/2.
-                    options.Protocols = HttpProtocols.Http2;
-
-                    if (string.Equals(uri.Scheme, "https", StringComparison.Ordinal))
-                    {
-                        options.UseHttps();
-                    }
+                    options.UseHttps();
                 }
             }
         }
     }
 
+    public Uri? DashboardServiceUri { get; init; }
+
+    private static bool TryGetDashboardServiceUriFromEnviroment([NotNullWhen(true)]out Uri? uri)
+    {
+        if (Environment.GetEnvironmentVariable(DashboardServiceUrlVariableName) is not { } value)
+        {
+            uri = null;
+            return false;
+        }
+
+        if (!Uri.TryCreate(value, UriKind.Absolute, out var candidateUri))
+        {
+            uri = null;
+            return false;
+        }
+
+        if (candidateUri.Scheme.ToLowerInvariant() != "https")
```

Comment:
Avoid a potential allocation:

```suggestion
        if (!StringComparer.InvariantCultureIgnoreCase.Equals(candidateUri.Scheme, "https"))
```